### ISSUE TITLE
H5Oget_info error with HDF library 1.10.3

### ISSUE THREAD

**Issue by BoundaryValueProblems** [JuliaIO/HDF5.jl#508]
Hello. It looks to me that HDF5.jl v0.10.0 is working well on Julia v0.7.0.
However, on Julia v0.6.4, I'm using HDF5.jl v0.9.4, and Pkg.test("HDF5") failed with the error:
```
julia> Pkg.test("HDF5")
INFO: Testing HDF5
HDF5 version 1.10.3
plain: Error During Test
  Got an exception of type ErrorException outside of a @test
  ccall: could not find function H5Oget_info in library /Users/xxx/.julia/v0.6/Homebrew/deps/usr/lib/libhdf5.dylib
```
Interesting enough, the HDF5 library used on v0.6.4 is 1.10.3 while that used on v0.7.0 is 1.10.2. I'm wondering if this is the source of this problem or not. (This might be the case since I ran ```Homebrew.update()``` on v0.6.4 while on v0.7.0, Homebrew.jl has problems and cannot be updated... Any help would be highly appreciated!

**Comment by BoundaryValueProblems** [comment]
I just wanted to add some more info. I just recovered ~/.julia/v0.6/Homebrew one week ago using Time Machine, which was the previous version of HDF5 library 1.10.2_1. Then, ```Pkg.test("HDF5")``` passed!  So, this is a warning. If you try to update the Homebrew under Julia v0.6.x, then the HDF5 library would be updated to 1.10.3 and you may encounter the same problem as I had. I recall when I updated Homebrew today, the HDF5 library was configured and compiled instead of copying the prebuilt library, which might have caused this problem although it installed the HDF5 library without any errors.

**Comment by BoundaryValueProblems** [comment]
Additional info:
* On HDF5 library v1.10.2, libhdf5.dylib contains the functions: H5Oget_info; H5Oget_info_by_idx; H5Oget_info_by_name.
* On HDF5 library v1.10.3, libhdf5.dylib contains the functions: H5Oget_info1; H5Oget_info2; H5Oget_info_by_idx1; H5Oget_info_by_idx2; H5Oget_info_by_name1; H5Oget_info_by_name2.
So, if we want to use the newer version of HDF5 library, then all the codes calling H5Oget_info must be replaced by H5Oget_info1 or 2 (I don't know yet what's the difference between them).

**Comment by jeanpauphilet** [comment]
Same issue here. Thanks for digging into it? Is there a easy way to come back to hdf5 1.10.2 with homebrew?

**Comment by BoundaryValueProblems** [comment]
@jeanpauphilet , before I ended up using Time Machine to recover the previous version, I googled how to downgrade a package in Homebrew, and got the following sites:
https://stackoverflow.com/questions/3987683/homebrew-install-specific-version-of-formula
https://jaketrent.com/post/downgrade-previously-installed-brew-formula/
https://zoltanaltfatter.com/2017/09/07/Install-a-specific-version-of-formula-with-homebrew/
But, these looked rather complicated, and some people reported failure. So, I just decided an easy solution: use Time Machine.
I wish HDF5.jl development team will address this problem very soon so that HDF5.jl works under either HDF5 library 1.10.2 or 1.10.3. HDF5.jl is an important package since JLD.jl, JLD2.jl, MAT.jl depend on this.
